server/backend/cron/news.go

```diff
@@ -137,39 +140,23 @@ func (c *CronService) parseNewsFeed(source model.NewsSource) error {
 	return nil
 }
 
-// saveImage Saves an image to the database so it can be downloaded by another cronjob and returns its id
-func (c *CronService) saveImage(url string) (null.Int, error) {
+// saveImage Saves an image to the database, so it can be downloaded by another cronjob
+func (c *CronService) saveImage(url string) (*model.Files, error) {
 	targetFileName := fmt.Sprintf("%x.jpg", md5.Sum([]byte(url)))
-	var fileId null.Int
-	if err := c.db.Model(model.Files{}).
-		Where("name = ?", targetFileName).
-		Select("file").Scan(&fileId).Error; err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
-		log.WithError(err).WithField("targetFileName", targetFileName).Error("Couldn't query database for file")
-		return null.Int{}, err
-	}
-	if fileId.Valid { // file already in database -> return for current news.
-		return fileId, nil
-	}
-
-	// otherwise store in database:
 	file := model.Files{
 		Name:       targetFileName,
 		Path:       ImageDirectory,
 		URL:        sql.NullString{String: url, Valid: true},
 		Downloaded: sql.NullBool{Bool: false, Valid: true},
 	}
-	err := c.db.Transaction(func(tx *gorm.DB) error {
-		if err := tx.Create(&file).Error; err != nil {
-			log.WithError(err).Error("Could not store new file to database")
-			return err
-		}
-		return nil
-	})
-	if err != nil {
-		return null.Int{}, err
+	if err := c.db.Clauses(clause.OnConflict{
+		Columns:   []clause.Column{{Name: "path"}},
+		DoNothing: true,
+	}).Create(&file).Error; err != nil {
+		log.WithError(err).WithField("targetFileName", targetFileName).Error("Couldn't query database for file")
+		return nil, err
 	}
-	// creating this int is annoying but i'm too afraid to use real ORM in the model
-	return null.Int{NullInt64: sql.NullInt64{Int64: int64(file.File), Valid: true}}, nil
```

Comment:
How is the file.File (the ID) actually populated here?

Reply:
You seemingly access that field outside of this function, but its never populated here.

Reply:
I reverted this part.
I don't know how but I am pretty sure that I checked if the filename is indeed an unique characteristic (apparetnly not properly)